src/msg.c

```diff
@@ -214,8 +214,7 @@ _canonicalKey(const char *key, char *keyStack, int keyStackLen, char **cKey, boo
     }
     else
     {
-        strncpy(keyStack, key, lenKey);
-        keyStack[lenKey] = '\0';
+        strlcpy(keyStack, key, lenKey + 1);
```

Comment:
First, let me say that I really appreciate your recent PRs since they improve the quality of the library.

For that one, however, I think that we can easily solve this without introducing all these changes.
The code here knows that the caller has a byte array that is big enough to contain the copy (+ terminating '\0'), otherwise, it would have gone to the preceding "if" statement and make a copy.

The compiler does not know that (unlike for the tests where the char [] is defined before the strncpy and compiler can "see" that the string copy clearly fit in the char [], hence you don't get the warnings for `test.c`).

So we could simply replace `strncpy` here with:
```
        memcpy((void*) keyStack, (const void*) key, (size_t) lenKey);
        keyStack[lenKey] = '\0';
        k = keyStack;
```